infra/terraform/modules/data_access/api_gateway.tf

```diff
@@ -0,0 +1,77 @@
+resource "aws_api_gateway_rest_api" "analytics" {
+  name = "${var.env}_analytics"
+}
+
+# API Gateway stage (to give access to API from the outside)
+resource "aws_api_gateway_deployment" "analytics" {
+  rest_api_id = "${aws_api_gateway_rest_api.analytics.id}"
+  stage_name  = "${var.env}"
```

Comment:
I'm not convinced the API stage should be named after the environment - it feels like it should be named `production` or even `api` - that way the stage names are the same in all environments, and represent their role in that environment specifically, rather than the environment they're part of.